src/download.py

```diff
@@ -113,16 +129,48 @@ async def download_file(self, url: str | URL) -> io.BytesIO:
         self.retries = 0
         return buf
 
-    def sync_download_file(self, url: str) -> io.BytesIO:
-        try:
-            return self.loop.run_until_complete(self.download_file(url))
-        except RuntimeError as e:
-            if e.args[0] == "Event loop is closed":
-                print("has to start a new event loop")
-                self.loop = asyncio.new_event_loop()
-                self._session = None
-                return self.loop.run_until_complete(self.download_file(url))
-            raise e
+    async def download_file_to_disk(self, url: str, path: str) -> None:
+        buf = await self.download_file(url)
+        # don't block the event loop for disk io
+        await self.loop.run_in_executor(
+            self.threadpool,
+            lambda: shutil.copyfileobj(buf, open(path, "wb"), length=2 << 18),
```

Comment:
Does it make sense to create the on-disk file object and not try and buffer the whole file and then write to disk, we could stream to disk with seek offsets. 

I know this a departure from pget, but also keep in mind you're allocating a bunch of memory, since this going to get heap allocated will it then be held onto through the runtime? With pget we have isolated memory allocation that will be fully released. Does this require an explicit `gc` call? Or is this going to be ok because the GC will allow for reuse when we begin loading the files into memory from disk regardless of free to the system?

I'm trying to think through the failure cases and these questions are intended to make sure we're not going to fall over due to some internal behavioral changes to the system.

Reply:
it's really bad in any case, engine should just load it from memory instead of roundtripping to disk, I'd rather spend time on that than figuring out seek 

buf should be deallocated once we leave download_file_to_disk, if we run into problems we can add a gc.collect() but it should eventually do the right thing  

a40-large has may more memory than it needs anyway

Reply:
As long as 72Gi is sufficient headroom for the file(s) and , sure lets not round-trip to disk. Is there any real usecase for disk-persistence through the lifecycle of the pod? (Before we say it's not relevant)

The answer to this absolutely can be "nothing at the moment".